joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -784,6 +785,34 @@ trait AstForExpressionsCreator(implicit withSchemaValidation: ValidationMode) {
     methodRef
   }
 
+  private def astForSingletonObjectMethodDeclaration(node: SingletonObjectMethodDeclaration): Ast = {
+    val methodName     = node.methodName
+    val methodFullName = computeSingletonObjectMethodFullName(s"class<<${node.baseClass.span.text}.$methodName")
+
+    val methodAstsWithRefs = node.block.body match {
+      case x: RubyBlock =>
+        astForMethodDeclaration(
+          x.toMethodDeclaration(methodName, Option(node.block.parameters)),
+          singletonMethodFullName = Option(methodFullName)
+        )
+      case _ =>
+        astForMethodDeclaration(
+          node.block.toMethodDeclaration(methodName, Option(node.block.parameters)),
+          singletonMethodFullName = Option(methodFullName)
+        )
+    }
+
+    // Set span contents
+    methodAstsWithRefs.flatMap(_.nodes).foreach {
+      case m: NewMethodRef => DummyNode(m.copy)(node.block.span.spanStart(m.code))
+      case _               =>
+    }
+
+    val Seq(_, methodRef) = methodAstsWithRefs
```

Comment:
From this PR https://github.com/joernio/joern/pull/4753, you'll want to emit the `typeRef` instead
```suggestion
    val Seq(typeRef, _) = methodAstsWithRefs
```